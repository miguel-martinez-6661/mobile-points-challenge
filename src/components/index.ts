export * from './text/text';
export * from './button/button';
export * from './card/card';
export * from './loading-indicator/loading-indicator';
