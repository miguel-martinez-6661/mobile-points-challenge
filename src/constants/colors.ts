// These colors should be defined in a design system and included in the theme file.
// This is just a quick example.
export enum Colors {
  primary = '#334FFA',
  secondary = '#CFD6FF',
  tertiary = '#9B9898',
  success = '#00B833',
  error = '#FF0000',
  background = '#FFFFFF',
  defaultBackground = '#f2f2f2',
  textPrimary = '#000000',
  textSecondary = '#FFFFFF',
  textTertiary = '#9B9898',
}
