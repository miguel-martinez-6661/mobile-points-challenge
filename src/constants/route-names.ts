export enum RouteNames {
  Home = 'Home',
  Details = 'Details',
}
