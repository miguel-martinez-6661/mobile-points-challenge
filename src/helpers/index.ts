export * from './numbers-helpers';
export * from './date-helpers';
