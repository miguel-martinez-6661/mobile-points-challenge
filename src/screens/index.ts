export * from './home/home-screen';
export * from './details/product-details-screen';
