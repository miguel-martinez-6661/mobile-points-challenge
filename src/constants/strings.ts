// This file contains all the strings used in the app.
// In case translations are needed, this is the where all keys are defined.
import { RouteNames } from './route-names';

export const strings = {
  [RouteNames.Home]: {
    greetings: 'Bienvenido de vuelta',
    yourPoints: 'TUS PUNTOS',
    yourPointsFlow: 'TUS MOVIMIENTOS',
    allButton: 'Todos',
    earnedPointsButton: 'Ganados',
    redeemedPointsButton: 'Canjeados',
  },
  [RouteNames.Details]: {
    productDetails: 'Detalles del producto',
    purchasePointsEarned: 'Con esta compra acumulaste',
    purchasedOn: 'Comprado el',
    points: 'puntos',
    acceptButton: 'Aceptar',
  },
};
