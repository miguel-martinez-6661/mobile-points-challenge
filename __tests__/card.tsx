import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { Card, Text } from '../src/components';
import { Colors } from '../src/constants/colors';

describe('Testing <Card> component', () => {
  it('should mount component correctly', () => {
    const cardComponent = renderer.create(
      <Card>
        <Text>Test</Text>
      </Card>,
    );
    expect(cardComponent).toMatchSnapshot();
  });

  it('should support custom styles by receiving a style objects as a prop', () => {
    const cardComponent = renderer
      .create(
        <Card styles={{ width: 50 }}>
          <Text>Test</Text>
        </Card>,
      )
      .toJSON();
    expect(cardComponent).toMatchSnapshot();
  });

  it('should be able to change background color by receiving a color as a prop', () => {
    const cardComponent = renderer
      .create(
        <Card color={Colors.secondary}>
          <Text>Test</Text>
        </Card>,
      )
      .toJSON();
    expect(cardComponent).toMatchSnapshot();
  });
});
