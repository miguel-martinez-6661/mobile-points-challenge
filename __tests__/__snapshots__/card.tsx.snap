// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Card> component should be able to change background color by receiving a color as a prop 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#334FFA",
        "borderRadius": 20,
        "elevation": 15,
        "minWidth": 432.1428571428571,
        "padding": 20,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
      },
      Object {
        "backgroundColor": "#CFD6FF",
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Avenir",
        "fontSize": 18.857142857142858,
        "fontWeight": "400",
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Testing <Card> component should mount component correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#334FFA",
        "borderRadius": 20,
        "elevation": 15,
        "minWidth": 432.1428571428571,
        "padding": 20,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
      },
      Object {
        "backgroundColor": "#334FFA",
      },
      Object {},
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Avenir",
        "fontSize": 18.857142857142858,
        "fontWeight": "400",
      }
    }
  >
    Test
  </Text>
</View>
`;

exports[`Testing <Card> component should support custom styles by receiving a style objects as a prop 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#334FFA",
        "borderRadius": 20,
        "elevation": 15,
        "minWidth": 432.1428571428571,
        "padding": 20,
        "shadowColor": "#000000",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 4,
      },
      Object {
        "backgroundColor": "#334FFA",
      },
      Object {
        "width": 50,
      },
    ]
  }
>
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Avenir",
        "fontSize": 18.857142857142858,
        "fontWeight": "400",
      }
    }
  >
    Test
  </Text>
</View>
`;
